Feature/12 create course list and course detail

PhucHoangVan opened the pull request:
- refactor file with index file
- fix comment clean
- fix function name camel case

PhucHoangVan committed:
#12 Add UI Course List Page, Component CourseItem

- UI for mobile
- UI for table
- UI for laptop
- UI for large screen
- Add fake data file
- Change route couse -> courseList

PhucHoangVan committed:
#12 refactor file and clean some code

- refactor file with index file
- fix comment clean
- fix function name camel case

PhucHoangVan committed:
#12 Create component CourseDetail

PhucHoangVan committed:
create child component for CourseDetail

- Chapter component
- Unit component
- Lesson component
- ModalRegisterCourse component

PhucHoangVan committed:
#12 Update CourseDetail

- Add fake data course for course detail
- update banner, course demo info

PhucHoangVan committed:
#12 move SubNav component

PhucHoangVan committed:
#12 Update CourseDetail - 2

- Insert fake data for Course Detail page (courseDetail, achieves, authors, targetUsers, skills and other infomation about this course)

- Add UI Achieve after complete course (achieve list displayed by number)

- Add UI course Infomation (General information about the course is displayed in this interface)

tuannt02 commented on src/features/course/components/CourseList/CourseList.jsx:
```diff
@@ -0,0 +1,106 @@
+
+      {/* Sub Navigation component*/}
+      <SubNav />
+
```
Component này nên để tên là khác. Ví dụ TabControl hay là gì đó .... Có thể tham khảo trên bootrap để đặt tên

tuannt02 commented on src/features/course/components/CourseList/CourseList.jsx:
```diff
@@ -0,0 +1,106 @@
+        {/* List Pro Course */}
+        {coursesPro?.length > 0 && (
+          <>
+            <div className="flex items-end mt-10">
```
Từ heading margin bot hoặc top (**mt-10**) có khoảng cách gần gần nhau, khó phân biệt. Nên cho heading của ông nào gần khối course item của ông đó. M để xa quá

tuannt02 reviewed (commented):
CourseList.jsx có lặp đi lặp lại một khối code Heading + Listitem. Nên tách ra thành một component nữa.

PhucHoangVan committed:
#12 Update courseDetail - 3

-  Insert fake data for Course Detail page (list chapter -> list unit -> list lesson)

- Add UI Course content (user can see list of chapters and lessons)

PhucHoangVan committed:
#12 Update courseDetail - 4

- Add UI component Chapter
- Add UI component Unit
- Add UI component Lesson

PhucHoangVan committed:
Add UI Modal Register Course

PhucHoangVan committed:
Add UI about course quality

PhucHoangVan committed:
#12 fix position modal Register

PhucHoangVan committed:
#12 handle when click open All

PhucHoangVan committed:
#12 fix UI for CourseList page

- fix UI
- detach component CourseListItem

PhucHoangVan committed:
#12 Detach component from CourseDetail

PhucHoangVan committed:
#12 Add component StarRating

tuan-hda committed:
Resolve conflict

tuan-hda committed:
Resolve conflicts
